cluster-autoscaler/processors/nodegroupset/label_nodegroups.go

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodegroupset
+
+import (
+	klog "k8s.io/klog/v2"
+	schedulerframework "k8s.io/kubernetes/pkg/scheduler/framework"
+)
+
+// CreateLabelNodeInfoComparator returns a comparator that checks for node group similarity using the given labels.
+func CreateLabelNodeInfoComparator(labels []string) NodeInfoComparator {
+	return func(n1, n2 *schedulerframework.NodeInfo) bool {
+		return areLabelsSame(n1, n2, labels)
+	}
+}
+
+func areLabelsSame(n1, n2 *schedulerframework.NodeInfo, labels []string) bool {
+	for _, label := range labels {
+		val1 := n1.Node().ObjectMeta.Labels[label]
+		val2 := n2.Node().ObjectMeta.Labels[label]
```

Comment:
What happens if the specified label is not present on either NodeInfo? Wouldn't this _panic_?

This seems like an easy use-case to hit if I'm using a custom `example.com/balance-group-name` label for when I have duplicated a nodegroup across AZs, but am not using it for nodegroups which are not set up that way as they are AZ-local for reasons other than EBS-selection. The latter would still be fed into this comparison function if they otherwise matched the pod-to-be-created.

And once this isn't panicking, should the behaviour of 'balancing label is not present' be:
* "Don't use this nodeset", i.e. the balancing should not even be attempted if the pod-match-chosen `n1` doesn't have the desired label, and if the potential-match `n2` lacks the label, it always returns false; or
* "Absence is also a value", i.e. only match if both NodeInfos lack the label, or if both have the label and the same value for that label; or
* something else?

Reply:
e.g., to implement the "missing label excludes from balancing completely", I guess it'd be:
```suggestion
		val1, ok1 := n1.Node().ObjectMeta.Labels[label]
		val2, ok2 := n2.Node().ObjectMeta.Labels[label]
		if !ok1 {
			klog.V(8).Infof("%s label not present on %s.", label, n1.Node().Name)
			return false
		}
		if !ok2 {
			klog.V(8).Infof("%s label not present on %s.", label, n2.Node().Name)
			return false
		}
```

Reply:
Either way, it'd be good to see the behaviour in the case of one, the other, or both node groups lacking the relevant label in the tests, so we know what was intended.

Reply:
Good point! My initial thinking is that if the label is missing completely, it wouldn't be be considered similar to another node group with also a missing label. The absence of the label would indicate "never consider this similar".

Reply:
I've added tests and your suggestion. I made a slight change to use an existence check instead of comparing on the empty string - so if a user does explicitly set an empty label value, it will be considered, allow users to use label keys alone as ways to group nodes.